tests/rptest/services/action_injector.py

```diff
@@ -0,0 +1,183 @@
+import dataclasses
+import random
+import time
+from threading import Thread
+from typing import Optional, List
+
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.utils.util import wait_until
+
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.failure_injector import FailureSpec, FailureInjector
+from rptest.services.redpanda import RedpandaService
+
+
+@dataclasses.dataclass
+class ActionConfig:
+    cluster_start_lead_time_sec: float
+    min_time_between_actions_sec: float
+    max_time_between_actions_sec: float
+    max_affected_nodes: Optional[int] = None
+
+    def random_time_in_range(self) -> float:
+        return random.uniform(self.min_time_between_actions_sec, self.max_time_between_actions_sec)
+
+
+class RandomNodeOp:
+    def __init__(self, redpanda: RedpandaService, config: ActionConfig, admin: Admin):
+        self.admin = admin
+        self.config = config
+        self.redpanda = redpanda
+        self.nodes_affected = set()
+
+    def limit_reached(self) -> bool:
```

Comment:
`limit_reached` is a very genetic name it's unclear which limit you refer to. maybe it makes sense to add a comment and/or rename it

Reply:
Thanks for the suggestion, I have renamed it to `max_affected_nodes_reached` and also added a docstring